t/22-svirt.t

```diff
@@ -298,12 +302,56 @@ subtest 'Method consoles::sshVirtsh::add_disk()' => sub {
+            $cmd = shift @last_ssh_commands;
+            like($cmd, qr/vmkfstools -v1 --deletevirtualdisk $exp_fullpath/, "Check name of image to be deleted");
+            $cmd = shift @last_ssh_commands;
+            like($cmd, qr%cp\s+/vmfs/volumes/nfs/hdd/$original_filename\s+$vmware_openqa_datastore*;%, "Copy hdd to $vmware_openqa_datastore");
+            $cmd = shift @last_ssh_commands;
+            like($cmd, qr/vmkfstools -v1 --clonevirtualdisk $vmware_openqa_datastore$original_filename --diskformat thin $exp_fullpath/, "Check size");
+
+            my $target_nodelist = $svirt->{domainxml}->findnodes('domain/devices/disk[@type="file" and @device="disk"]/target[@dev="hd' . $dev_id . '" and @bus="ide"]');
+            is($target_nodelist->size, 1, 'Only one <target> with that dev_id exists');
+            my $target_node = $target_nodelist->shift;
+            is(defined($target_node), 1, '<target> node exists');
+            my $source_node = $target_node->parentNode->findnodes('source')->shift;
+            is($source_node->getAttribute('file'), '[my_vmware_datastore] openQA/' . $exp_filename, 'XML dokument contains correct file set');
+        };
+        subtest 'family vmware backingfile=1 xz file' => sub {
+            my $dev_id = 'dev_id_002a';
+            @last_ssh_commands = ();
+            @ssh_cmd_return    = (0, 0);
+            my $file         = "my_compressed_hdd.vmdk.xz";
+            my $vm_name      = $svirt->name;
+            my $exp_filename = "${vm_name}${dev_id}.vmdk";
+            set_var(VMWARE_NFS_DATASTORE => 'nfs');
+            my $exp_fullpath_xz = $vmware_openqa_datastore . $file;
+            my $exp_fullpath    = $vmware_openqa_datastore . $exp_filename;
+            $svirt->add_disk({
+                    backingfile => 1,
+                    dev_id      => $dev_id,
+                    file        => '/my/path/to/this/file/' . $file,
+            });
+
+            my $cmd = shift @last_ssh_commands;
+            like($cmd, qr@\( set -x; vmid=\$\(vim-cmd vmsvc/getallvms \| awk '/$vm_name/ \{ print \$1 \}'\);if \[ \$vmid \]; then vim-cmd vmsvc/power\.off \$vmid;vim-cmd vmsvc/destroy \$vmid;fi; sleep 10 \) 2>&1@, "Check delete and destroy");
+            $cmd = shift @last_ssh_commands;
+            like($cmd, qr/vmkfstools -v1 --deletevirtualdisk $exp_fullpath/, "Check name of image to be deleted");
+            $cmd = shift @last_ssh_commands;
+            like($cmd, qr%cp\s+/vmfs/volumes/nfs/hdd/$file\s+$vmware_openqa_datastore*;%, "Copy hdd to $vmware_openqa_datastore");
+            $cmd = shift @last_ssh_commands;
+            like($cmd, qr%xz --decompress --keep --verbose --no-warn --quiet '$exp_fullpath_xz'%, 'Uncompress file with unxz');
+            $cmd = shift @last_ssh_commands;
+            like($cmd, qr/vmkfstools -v1 --clonevirtualdisk $vmware_openqa_datastore$file --diskformat thin $exp_fullpath/, "Check size");
```

Comment:
I guess we need to use `$file` without `.xz` extention.

Reply:
yes, already fixed :D